api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```diff
@@ -0,0 +1,251 @@
+            saveReportDesigns(reportDefinition, reportDesigns);
+        }
+    }
+
+    public static void saveReportDefinition(ReportDefinition reportDefinition) {
+
+        ReportDefinition existing = Context.getService(ReportDefinitionService.class).getDefinitionByUuid(reportDefinition.getUuid());
+        if (existing != null) {
+            // we need to overwrite the existing, rather than purge-and-recreate, to avoid deleting old ReportRequests
+            reportDefinition.setId(existing.getId());
+        }
+        else {
+            // incompatible class changes for a serialized object could mean that getting the definition return null
+            // and some serialization error gets logged. In that case we want to overwrite the invalid serialized definition
+            SerializedObject invalidSerializedObject = Context.getRegisteredComponents(SerializedObjectDAO.class).get(0).getSerializedObjectByUuid(reportDefinition.getUuid());
+            if (invalidSerializedObject != null) {
+                reportDefinition.setId(invalidSerializedObject.getId());
+            }
+        }
+
+        Context.getService(ReportDefinitionService.class).saveDefinition(reportDefinition);
+    }
+
+    public static void saveReportDesigns(ReportDefinition reportDefinition, List<ReportDesign> reportDesigns) {
+        // purging a ReportDesign doesn't trigger any extra logic, so we can just purge-and-recreate here
+        List<ReportDesign> existingDesigns = Context.getService(ReportService.class).getReportDesigns(reportDefinition, null, true);
+        if (existingDesigns.size() > 0) {
+            log.debug("Deleting " + existingDesigns.size() + " old designs for " + reportDefinition.getName());
+            for (ReportDesign design : existingDesigns) {
+                Context.getService(ReportService.class).purgeReportDesign(design);
+            }
+        }
+
+        for (ReportDesign reportDesign : reportDesigns) {
+            Context.getService(ReportService.class).saveReportDesign(reportDesign);
+        }
+
+    }
+
+    public static  ReportDefinition constructReportDefinition(ReportDescriptor reportDescriptor) {
+
+        ReportDefinition rd = new ReportDefinition();
+        // TODO do we need to handle translations here?
+        rd.setName(reportDescriptor.getName());
+        rd.setDescription(reportDescriptor.getDescription());
+        rd.setUuid(reportDescriptor.getUuid());
+
+        List<Parameter> parameters = constructParameters(reportDescriptor.getParameters());
+        Map<String, Object> mappings = constructMappings(parameters);
+        rd.setParameters(parameters);
+
+        if (reportDescriptor.getDatasets() != null) {
+            for (DataSetDescriptor dataSetDescriptor : reportDescriptor.getDatasets()) {
+                DataSetDefinition dsd = constructDataSetDefinition(dataSetDescriptor, parameters);
+                // TODO do we need to handle DSD name and description?
```

Comment:
The answer is no.  Name and description are only really intended for when a definition is something saved at the top level, it's fine to leave blank for a DSD that is nested and saved within a ReportDefinition